### PR TITLE
Ian's NER Tagger frame

### PR DESCRIPTION
**Pull request by iajohns1**
This frame implements:
- Basic interface
- tagger_service() which calls the tagger service and returns its results (mock)
- results_to_keywords() which converts tagger results to the keyword object (functional)
- get_tags() which pulls the functions together and returns an array of keywords from a search string

### CODE REVIEW

File: test/spec/SearchApi/Providers/NerTaggerSpec.php~ [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<?php
```
Comment (idmontie) [inline test/spec/SearchApi/Providers/NerTaggerSpec.php~]: Ignore this file.

File: prepend.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
```
Comment (rraub) [inline prepend.php]: Is this your testing file? if so then it probably shouldn't be committed

File: src/models/keyword.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * @var	$text		string	The actual keyword
+ * @var	$type		string	The type of info the keyword represents
+ * @var	$relevance	double	The improtance of the keyword (0.0 - 1.0)
+ */
+class Keyword {
+  public $text;
+  public $type;
+  public $relevance;
+
+  function __construct( $a1, $a2, $a3 ) {
```
Comment (idmontie) [inline src/models/keyword.php]: NO!

:-1:

File: src/providers/ner-tagger.php [hunk 4]
```diff
@@ -1,12 +1,71 @@
...
+    return $keywords;
+  }
+
+  // Call the NER service and save results as a string containing an XML document
+  public function tagger_service( $request_string = '' ) {
+    if ( $request_string === null || $request_string === '' ) {
+      return null;
+    }
+
+    $tagger_results = array(
```
Comment (rraub) [inline src/providers/ner-tagger.php:30]: Rework this to this kind of format:

``` [
 ['Phoenix', 'LOCATION'],
 ['Arizona', ....
```

Also add a comment saying that this is test data

File: test/spec/SearchApi/Providers/NerTaggerSpec.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    $this->get_tags( null )->shouldReturn( null );
+  }
+
+  // The tagger service should return a nested array object
+  function it_should_return_nested_array() {
+    // Test a single function call instead of several
+    $test_result = $this->tagger_service( 'test' );
+
+    $test_result->shouldBeArray();
+    foreach ( $test_result as &$result ) {
```
Comment (rraub) [inline test/spec/SearchApi/Providers/NerTaggerSpec.php]: dont need to pass by ref

File: src/providers/ner-tagger.php [hunk 6]
```diff
@@ -1,12 +1,71 @@
...
+    $tagger_results = $this->tagger_service( $request_string );
+    // Convert results into keywords
+    $keywords = $this->results_to_keywords( $tagger_results );
+
+    return $keywords;
+  }
+
+  // Call the NER service and save results as a string containing an XML document
+  public function tagger_service( $request_string = '' ) {
+    if ( $request_string === null || $request_string === '' ) {
```
Comment (rraub) [inline src/providers/ner-tagger.php]: You should use php's [empty](https://secure.php.net/manual/en/function.empty.php)() function here

Comment (idmontie) [inline src/providers/ner-tagger.php]: :+1: `if ( empty( $request_string )`

Comment (rraub) [inline src/providers/ner-tagger.php]: This is the last thing I would like to see changed, otherwise its good to merge!

**Pull request merged**